### DISCUSSION TITLE
Trip definition in recreational data

### DISCUSSION THREAD

**Discussion by brianlangseth-NOAA** [pfmc-assessments/canary_2023#55] [Q&A] [answered]
This is a question related to composition data and only affects weighting of data sources within the model. 

Ive calculated the number of trips to use as sample size for recreational comps. We have used number of fish in the past, but fish caught within the same location are probably more similar to one another, so trip can be used to get at a measure of "unique" samples. 

Below is what I used to define trip. These are based similarly on what Copper rockfish has used, but that is California centric, and I want to ensure Im not missing wildly on this. Are these reasonable fields to approximate the number of trips? 

California (@JohnBudrick)
RecFIN: day (RECFIN_DATE), location (COUNTY_NUMBER and AGENCY_WATER_AREA_NAME), mode (RECFIN_MODE_NAME) 
MRFSS: time (Year), id (ID_CODE), location (INTSITE and AREA_X), mode (MODE_FX)

Washington (@tsoutt, @KSten)
~sport data: time (YEAR), id (sequence_number), location (mfbds_v_sample.punch_card_area_code - the only one that isn't NA)~
sport data: time (fish_sample_date OR if blank then sample_date), location (mfbds_v_sample.punch_card_area_code OR if blank then mfbds_v_fish.punch_card_area_code), mode

Oregon (@aliwhitman)
state provided recfin: time (RECFIN_DATE), location (RECFIN_PORT_NAME), id (ANGLER_ID) 
state provided mrfss: time (Year), id (id_code), location (ORBS_Port and MRFSS_AREA_X), mode (Mode_FX_Name)
state provided age (and lengths with ages): per advice from Ali (and consistent with black rockfish): time (SAMPLE_DATE), location (PORT_NAME), and mode

**Comment by aliwhitman** [comment]
ANGLER_ID is the only field needed to delineate a trip with the OR RecFIN data. 

And id_code is the only field needed to identify a trip in the OR MRFSS data.  So even easier than you think!  

I would note that if we are going to use number of trips for the OR age comp data (which is coming from a different table in RecFIN), I will need to combine some fields to create a trip identifier.  But it can be done, just noting that it will be different than the length comp data already provided.

**Reply by brianlangseth-NOAA** [reply]
Thanks Ali. If the necessary fields are already within the age csv you provide I can combine them myself into a trip. If not, then providing a trip identifier would be helpful.

**Reply by brianlangseth-NOAA** [reply]
Ali - checking on this now. I see that MRFSS_AREA_X, ORBS_port, and mode provide slightly more resolution for mrfss data than just id_code. There are about 90 more 'trips' if I include all of those. What this means is that if I use only id_code, then Im combing samples among 90 trips that have some combination of different Ports, modes, MRFSS areas. Shall I proceed with just id_code (i.e can the same trip have a different port and mode)? 

Recfin data are the same when only have angler_id - so good there.

**Reply by aliwhitman** [reply, marked as answer]
So this is a little concerning for me, as all of my documentation for MRFSS states that id code is a trip identifier.  However, it's not really possible to have more than one mode (charter/private) on a trip, because that would be two different vessels. I can see more than one area (a combined estuary/ocean trip - these are quite common) and ORBS ports are assigned post-hoc to MRFSS data based on the interview site description, so those can occasionally be a bit strange.  So those last two are not super concerning.  

At this point, I would recommend using the combination of Area, ORBS port and mode, since multiple modes doesn't really make sense to define a trip.

**Comment by brianlangseth-NOAA** [comment]
@JohnBudrick and @tsoutt, @KSten - I forgot to bring up today but if I should be using a different definition of "trip" than what I show above in my original comment please let me know.

**Reply by KSten** [reply]
Hi @brianlangseth-NOAA.  WA can be a little complicated for Rec and it looks like I didn't provide the fish_sample_date field which may have been helpful.

In recent years (2017-2022) the issue of using sequence number is that it’s not being represented the same way it was in the past.  In years past, each sampling event with an individual date went in as an individual sequence number.  Now those individual sampling events are lumped together throughout a month as best as they can to make a 100 fish sample (that's how many otoliths fit in a tray).  So in theory, even though we are sampling more individual events, it’s not being represented the way it would have been prior to 2017.  And with Canary bring a prohib for quite some time prior to that, maybe we are underepresenting the actual number of "trips"?

Overall, year/sequence/PunchCardArea should be okay, but if you want a more discrete count, I can provide the fish sample date as well.

You also might need to grab the Punch Card Area from two different places, "mfbds_v_sample.punch_card_area_code" and if that is blank, "mfbds_v_fish.punch_card_area_code"  The perks of starting to collect data on the fish level instead of the sample level!

Hopefully that all makes sense.  I'm on the water this week through possibly next Wednesday if the weather holds out, but I can work at getting you more information if you'd like, or if you're happy with your current definition of "trip", I'm not opposed to it either, I just wanted to bring up the change.

**Reply by brianlangseth-NOAA** [reply]
@Ksten - thank you for the information. Having a definition of a trip change over time is of concern, so I would like to change out using sequence number. As of March 28 (though correct me if Im wrong), you were waiting for updated ages within the sport bio data. If you add fish-sample-date when you update the data that would be helpful. 

mfbds_v_fish.punch_card_area_code is all blank. In fact a lot of the area fields are blank. mfbds_v_sample.punch_card_area_code is the only area specific field that is populated, but ~40% of the time it is blank too, and all of these occur after 2019.  That is concerning since recent data would then falsely reflect fewer trips. 

Unless you also didn't provide some of the area fields, I may revert to fish as the effective sample number for WA. If you can populate at least one area field (like mfbds_v_fish.punch_card_area_code or port_name) then I would feel comfortable using the following to help approximate number of trips

time (fish-sample-date), location (mfbds_v_sample.punch_card_area_code, mfbds_v_fish.punch_card_area_code, or perhaps port_name)

**Reply by KSten** [reply]
@brianlangseth-NOAA  Yes, we are still waiting for final ages to get entered.  When I pull everything, I will be sure to add the fish sample date.  I'm happy to help make sense of all of it when I do that pull.

As for the catch area.  In the data I provided in the Drive, when "mfbds_v_sample.punch_card_area_code" is null, "mfbds_v_fish.punch_card_area_code" is filled in all cases but one fish, which has a port of La Push, so I left it as I assume it is Marine Area 3 with that port.  This is due to collecting the Marine Area at the fish level rather than the header level.  If you are pulling data from RecFIN, the fish level data is not delivered to them yet, we've asked but it's still not been implemented, so that would be the reason for empty Marine Areas.

**Reply by brianlangseth-NOAA** [reply]
@KSten I figured out why I was confused, my read_excel call wasn't registering the values in fish.punch_card_area_code. Ive fixed that and am tracking with you. I'm all set on how to calculate trip (fish-sample-date and a combination of sample.punch_card and fish.punch_card area codes). I will wait for the fish-sample-dates once you update the ages.

**Reply by brianlangseth-NOAA** [reply]
All set with age data and trip calcs. Ive added mode in my definition as well which is consistent with how we've combined for other data sources.

**Comment by brianlangseth-NOAA** [comment]
Closes with 4e3945b
